addons/sale/models/sale_order.py

```diff
@@ -743,6 +744,14 @@ def _onchange_company_id_warning(self):
                 }
             }
 
+    @api.onchange('company_id')
+    def _inverse_company_id(self):
+        for order in self:
+            # This can't be caught by a python constraint as it is only triggered at save and the compute method that
+            # needs this data to be set correctly before saving
+            if not order.company_id:
+                raise ValidationError(_("We can't leave this document without any company. Please select a company for this document."))
```

Comment:
Hi @myhy-odoo I didn't look thoroughly at the PR, but this requires a pot update in stable.